### DISCUSSION TITLE
Update markers that survive YAML generation

### DISCUSSION THREAD

**Discussion by squaremo** [fluxcd/flux2#1207] [Q&A] [answered]
https://github.com/fluxcd/image-automation-controller/issues/134 brings up the question of how to put update markers in generated files. Since YAML serialisation will usually elide comments, programs tend to omit or remove update markers.

One possibility is to use annotations, since these are YAML data and will generally survive processing.

**Comment by toddkazakov** [comment, marked as answer]
I created [kyaml-a2c](https://github.com/toddkazakov/kyaml-a2c/), a command line utility that converts annotations to markers that can be used as a post-processing step in manifest generation pipelines. The readme explains how it can be used.
